libs/network/include/network/service/promise.hpp

```diff
@@ -215,29 +212,26 @@ class PromiseError final : public std::exception
 };
 
 template <typename T>
-T details::PromiseImplementation::As() const
+bool details::PromiseImplementation::GetResult(T &ret) const
 {
-  T result{};
-  if (!As<T>(result))
-  {
-    throw PromiseError{*this};
-  }
+  bool success{false};
 
-  return result;
-}
+  try
+  {
+    if (Wait())
+    {
+      SerializerType ser(value_);
+      ser >> ret;
 
-template <typename T>
-bool details::PromiseImplementation::As(T &ret) const
-{
-  if (!Wait())
+      success = true;
+    }
+  }
+  catch (std::exception const &ex)
   {
-    return false;
+    FETCH_LOG_WARN("Promise", "Error getting promise result: ", ex.what());
```

Comment:
So now we won't see the service id etc in the reactor catch as a side effect of this right?

Reply:
Good spot this is fixed in the latest updates